x-pack/plugins/stack_connectors/server/connector_types/openai/openai.ts

```diff
@@ -114,6 +114,8 @@ export class OpenAIConnector extends SubActionConnector<Config, Secrets> {
       method: 'post',
       responseSchema: RunActionResponseSchema,
       data: sanitizedBody,
+      // give up to 2 minutes for response
+      timeout: 120000,
```

Comment:
Have you tried this in practice?  The thing I'm worrying about is that task manager could potentially "kill" the task if it hit's IT's timeout limit - which I believe can be extended but will need to look up how ...

TM can't really kill a task, but will basically ignore it if it runs over it's time-limit, and would then generate some kind of warning / error.  And _potentially_ retry it, but I believe it doesn't today ...

Reply:
See this issue https://github.com/elastic/kibana/issues/166561

Reply:
Also see this issue! https://github.com/elastic/security-team/issues/7873

Reply:
I think the default task timeout is 5 minutes and I don't think we set a custom timeout for action tasks so this shouldn't be cancelled due to timeout